batch/src/main/scala/uk/gov/ons/addressindex/utils/SqlHelper.scala

```diff
@@ -309,28 +309,42 @@ object SqlHelper {
+
+        val nisraStreet: Option[String] = Try(outputNisra.headOption.get("thoroughfare").toString).toOption
+        val nisraTown: Option[String] = Try(outputNisra.headOption.get("townName").toString).toOption
+        val nisraStart: Option[String] = outputNisra.headOption.flatMap(_.get("mixedNisraStart").map(_.toString))
+
         val lpiStreet: Option[String] = outputLpis.headOption.flatMap(_.get("streetDescriptor").map(_.toString))
+        val lpiStreetEng: Option[String] = EnglishNag.map(_.get("streetDescriptor").map(_.toString).getOrElse(""))
         val pafStreet: Option[String] = outputPaf.headOption.flatMap(_.get("thoroughfare").map(_.toString))
         val lpiTown: Option[String] = outputLpis.headOption.flatMap(_.get("townName").map(_.toString))
+        val lpiTownEng: Option[String] = EnglishNag.map(_.get("townName").map(_.toString).getOrElse(""))
+        val lpiLocality: Option[String] = outputLpis.headOption.flatMap(_.get("locality").map(_.toString))
+        val lpiLocalityEng: Option[String] = EnglishNag.map(_.get("locality").map(_.toString).getOrElse(""))
         val pafTown: Option[String] = outputPaf.headOption.flatMap(_.get("postTown").map(_.toString))
-        val nisraStreet: Option[String] = Try(outputNisra.headOption.get("thoroughfare").toString).toOption
-        val nisraTown: Option[String] = Try(outputNisra.headOption.get("townName").toString).toOption
+        val pafDepend = outputPaf.headOption.flatMap(_.get("dependentLocality").map(_.toString))
 
-        val nisraStart: Option[String] = outputNisra.headOption.flatMap(_.get("mixedNisraStart").map(_.toString))
-        val pafStart: Option[String] = outputPaf.headOption.flatMap(_.get("mixedPafStart").map(_.toString))
+        val lpiStartEng: Option[String] = EnglishNag.map(_.get("mixedNagStart").map(_.toString).getOrElse(""))
         val lpiStart: Option[String] = outputLpis.headOption.flatMap(_.get("mixedNagStart").map(_.toString))
-
         val bestStreet: String = if (!nisraStreet.getOrElse("").isEmpty) nisraStreet.getOrElse("")
         else if (!pafStreet.getOrElse("").isEmpty) pafStreet.getOrElse("")
+        else if (!lpiStreetEng.getOrElse("").isEmpty) lpiStreetEng.getOrElse("")
         else if (!lpiStreet.getOrElse("").isEmpty) lpiStreet.getOrElse("")
-        else if (!nisraStart.getOrElse("").isEmpty) "(" + lpiStart.getOrElse("") + ")"
+        else if (!nisraStart.getOrElse("").isEmpty) "(" + nisraStart.getOrElse("") + ")"
+        else if (!lpiStartEng.getOrElse("").isEmpty) "(" + lpiStartEng.getOrElse("") + ")"
         else "(" + lpiStart.getOrElse("") + ")"
 
         val bestTown: String = if (!nisraTown.getOrElse("").isEmpty) nisraTown.getOrElse("")
+        else if (!pafDepend.getOrElse("").isEmpty) pafDepend.getOrElse("")
+        else if (!lpiLocalityEng.getOrElse("").isEmpty) lpiLocalityEng.getOrElse("")
+        else if (!lpiLocality.getOrElse("").isEmpty) lpiLocality.getOrElse("")
```

Comment:
going forward.... it could be worth looking at option chaining for some of these cases such as 
nisraStreet **orElse**
pafStreet **orElse**
lpiStreetEng **orElse**
lpiStreet

Reply:
agree orElse would look cleaner